test/deployables/strategies/StrategyV1.test.ts

```diff
@@ -0,0 +1,702 @@
+      await strategy.connect(owner).addAdapter(await mockAdapter1.getAddress());
+      await strategy.connect(owner).addAdapter(await mockAdapter2.getAddress());
+
+      // mockAdapter1 says NO, mockAdapter2 says YES
+      await mockAdapter1.connect(owner).setProposerStatus(user1.address, false);
+      await mockAdapter2.connect(owner).setProposerStatus(user1.address, true);
+
+      void expect(await strategy.isProposer(user1.address)).to.be.true;
+    });
+
+    it('should return false if no adapter identifies the address as a proposer', async () => {
+      await strategy.connect(owner).addAdapter(await mockAdapter1.getAddress());
+      await strategy.connect(owner).addAdapter(await mockAdapter2.getAddress());
+
+      await mockAdapter1.connect(owner).setProposerStatus(user1.address, false);
+      await mockAdapter2.connect(owner).setProposerStatus(user1.address, false);
+
+      void expect(await strategy.isProposer(user1.address)).to.be.false;
+    });
+
+    it('should return true if the first adapter identifies the address as a proposer', async () => {
+      await strategy.connect(owner).addAdapter(await mockAdapter1.getAddress());
+      await strategy.connect(owner).addAdapter(await mockAdapter2.getAddress());
+
+      await mockAdapter1.connect(owner).setProposerStatus(user1.address, true);
+      await mockAdapter2.connect(owner).setProposerStatus(user1.address, false); // Should not be checked
+
+      void expect(await strategy.isProposer(user1.address)).to.be.true;
+    });
+  });
+
+  describe('getVotingTimestamps & getProposalBlocks', () => {
+    let proposalId: number;
+    let encodedProposalId: string;
+
+    beforeEach(async () => {
+      proposalId = 1;
+      encodedProposalId = ethers.AbiCoder.defaultAbiCoder().encode(['uint32'], [proposalId]);
+      // Ensure at least one adapter is added for initializeProposal to succeed
+      await strategy.connect(owner).addAdapter(await mockAdapter1.getAddress());
+    });
+
+    it('should return correct timestamps after proposal initialization', async () => {
+      const blockBefore = await ethers.provider.getBlock('latest');
+      if (!blockBefore) throw new Error('Failed to get latest block');
+      const timestampBefore = blockBefore.timestamp;
+
+      await strategy
+        .connect(azoriusMock)
+        .initializeProposal(encodedProposalId, [], ethers.ZeroHash);
+
+      const [startTime, endTime] = await strategy.getVotingTimestamps(proposalId);
+
+      expect(startTime).to.be.closeTo(timestampBefore + 1, 2);
+      expect(endTime).to.be.closeTo(timestampBefore + 1 + DEFAULT_VOTING_PERIOD, 2);
+    });
+
+    it('getVotingTimestamps should revert if proposal is not initialized', async () => {
+      const uninitializedProposalId = 999;
+      await expect(
+        strategy.getVotingTimestamps(uninitializedProposalId),
+      ).to.be.revertedWithCustomError(strategy, 'ProposalNotInitialized');
+    });
+  });
+
+  describe('isPassed', () => {
```

Comment:
Missing a test case to verify that block.timestamp <= proposal.votingEndTimestamp?

Reply:
indeed. a test for this is added in the next PR though.